drivers/esp-hosted/esp_hosted.proto

```diff
@@ -0,0 +1,432 @@
+syntax = "proto3";
```

Comment:
This file needs a license and copyright.  Can you add the MIT header here?

Reply:
This file defines the protobuf protocol, it's hosted here:

https://github.com/espressif/esp-hosted/blob/master/esp_hosted_fg/common/proto/esp_hosted_config.proto

It doesn't have any license, which license to add MicroPython ?

Reply:
Since this is copied from that repo linked above we cannot add our license/copyright (I thought you wrote this yourself).

It would be good to understand what the license of this file is (GPL 2 or Apache 2, it's not clear from that repo), probably need to open an issue on that repo to ask that question.

It's also not clear what the license/copyright is for the generated protobuf .c and .h files.  I would guess it's the same as this file.

Reply:
Would it be easier to just remove the protocol specification file ? 

> It's also not clear what the license/copyright is for the generated protobuf .c and .h files

I think we own the auto-generated file (protobuf's license doesn't apply to it either, I checked).

Reply:
IANAL but the auto-generated output is likely a derived work of the source file (this .proto file), so the same license would apply (in the same way that you can't compile or pre-process someone's copyrighted source code and claim that you now own the compiler output.)

@ameyinamdar @igrr Are you folks able to clarify the license for this .proto file in the esp-hosted repo, please?

Reply:
@projectgus They are dual-licensed under Apache 2.0 and GPL 2.0 (at your option). We will clarify this in esp-hosted!

Reply:
@dpgeorge License has been updated, does this work for you ?

```
 /* Copyright (C) 2015-2023 Espressif Systems (Shanghai) PTE LTD */
/* SPDX-License-Identifier: Dual-licensed under GPL-2.0 or Apache-2.0, at your option */
```

Reply:
> `/* SPDX-License-Identifier: Dual-licensed under GPL-2.0 or Apache-2.0, at your option */`

That's actually not a valid SPDX-License-Identifier, the old one (with `OR`) was the right one. Sorry about that, will fix.

Reply:
Thanks @igrr for the quick response, much appreciated! :)

Reply:
Considering we have an option here, can we explicitly select the Apache option and write:
```
SPDX-License-Identifier: Apache-2.0
```
AFAIK we are in full compliance with the license by doing this, because we have chosen a specific license and referenced that license.  If someone wanted this file as GPL then they can go back to the source and copy it with the GPL license specifier.

The reason to do this is to make it clear which license we have chosen.

Reply:
Updated, thanks!